### PR TITLE
Try fix vertx set timeout oom and reduce logging bursts

### PR DESCRIPTION
**Pull request by hiddenalpha**
Related: SDCISA-16293

### CODE REVIEW

File: src/main/java/org/swisspush/redisques/performance/BurstSquasher.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+import io.vertx.core.Vertx;
+
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+import static java.lang.Math.max;
+import static java.lang.System.currentTimeMillis;
+
+public class BurstSquasher<Ctx> {
```
Comment (mcweba) [inline src/main/java/org/swisspush/redisques/performance/BurstSquasher.java:11]: Please add some javadoc to document what this class does

File: src/main/java/org/swisspush/redisques/performance/UpperBoundParallel.java [hunk 2]
```diff
@@ -92,10 +95,15 @@ private <Ctx> void resume(Request<Ctx> req) {
                     }
                     return;
                 }
-                if (!req.limit.tryAcquire()) {
+                if (req.numTokensAvailForOurself > 0) {
+                    // We still have a token reserved for ourself. Use those first before acquiring
+                    // new ones. Explanation see comment in 'onOneDone()'.
+                    req.numTokensAvailForOurself -= 1;
+                }else if (!req.limit.tryAcquire()) {
```
Comment (dominik-cnx) [inline src/main/java/org/swisspush/redisques/performance/UpperBoundParallel.java:102]: trivial, code format

File: src/main/java/org/swisspush/redisques/performance/UpperBoundParallel.java [hunk 3]
```diff
@@ -116,8 +131,19 @@ private <Ctx> void resume(Request<Ctx> req) {
...
-                vertx.setTimer(RETRY_DELAY_IF_LIMIT_REACHED_MS, nonsense -> resume(req));
+                if (!req.hasStarted) {
+                    // We couldn't even trigger one single task. No resources available to
+                    // handle any more requests. This caller has to try later.
+                    req.isFatalError = true;
+                    Exception ex = exceptionFactory.newResourceExhaustionException(
+                            "No more resources to handle yet another request now.", null);
+                    req.mentor.onError(ex, req.ctx);
+                    return;
+                }else{
```
Comment (dominik-cnx) [inline src/main/java/org/swisspush/redisques/performance/UpperBoundParallel.java:142]: code format

**Review by linovifian — APPROVED** [review]

File: src/main/java/org/swisspush/redisques/performance/UpperBoundParallel.java (hunk 2 above)
Comment (hiddenalpha) [reply src/main/java/org/swisspush/redisques/performance/UpperBoundParallel.java:102]: If ok, I will do minor stuff in a separate PR.

**Review by dominik-cnx — APPROVED** [review]

**Pull request merged**